common/views/components/ButtonSolid/ButtonSolid.tsx

```diff
@@ -106,18 +106,21 @@ export enum ButtonTypes {
   submit = 'submit',
 }
 
+type ButtonSize = 'small' | 'medium' | 'large';
+
 export type ButtonSolidBaseProps = {
   text: ReactNode;
   icon?: IconSvg;
   type?: ButtonTypes;
   isTextHidden?: boolean;
   trackingEvent?: GaEvent;
-  isBig?: boolean;
   ariaControls?: string;
   ariaExpanded?: boolean;
   ariaLive?: 'off' | 'polite' | 'assertive';
   colors?: ButtonColors;
   isIconAfter?: boolean;
+  size?: ButtonSize;
```

Comment:
For my own knowledge of TS declarations; I noticed the padding starts at `13px 20px` (medium sized), but then gets redeclared in the CSS if the prop `size: medium` is called. Wouldn't it also be an option here to make it default to `medium` if not specified?

Reply:
Yes, I think this makes sense. I think it'll be a bit fiddly because `BaseButton` is being exported and there's already quite a bit of (probably/maybe unnecessary prop-drilling from the React component to the styled components. I'll pick it up next week.

Reply:
Styled-components `defaultProps` and TypeScript [don't work great together](https://stackoverflow.com/a/52249198/590593). I've removed the padding from the `BaseButton` though (since it was already being overwritten where it was used elsewhere). And then gave a default (`medium`) value to a function that creates the padding, which felt like a reasonable middle ground to me. I think an alternative would have been to destructure the props inside the styled-component and add a default value to `size` at that point, but I found the syntax within back-ticks to be less expressive and I found it harder to see what was going on.